lib/appsignal/span.ex

```diff
@@ -190,18 +190,23 @@ defmodule Appsignal.Span do
       |> Appsignal.Span.set_sample_data("environment", %{"method" => "GET"})
 
   """
-  def set_sample_data(%Span{reference: reference} = span, key, value)
-      when is_binary(key) and is_map(value) do
-    data =
-      value
-      |> Appsignal.Utils.MapFilter.filter()
-      |> Appsignal.Utils.DataEncoder.encode()
+  def set_sample_data(span, "params" = key, value) do
+    do_set_sample_data(span, key, Appsignal.Utils.MapFilter.filter(value))
```

Comment:
When matching on `"params"`, we can use `"params"` in the function clause’s body instead of putting it into a variable.